Documentation/Books/Manual/Programs/Arangodump/Maskings.md

```diff
@@ -88,14 +109,140 @@ or
 
     "path": "`name.with.dots`"
 
-xify_front
-----------
+If the attribute value is an array the masking is applied to all the
+array elements.
+
+### Example
+
+This following will replace name by a "XXXX" masked string
+
+    {
+      "type": "xify_front",
+      "path": ".name",
+      "unmaskedLength": 2
+    }
+
+The object
+
+    {
+      "name": "top-level-name",
+      "age": 42,
+      "nicknames" : [ { "name": "hugo" }, "egon" ],
+      "other": {
+        "name": [ "emil", { "secret": "superman" } ]
+      }
+    }
+
+will be replaced as follows:
+
+    {
+      "name": "xxxxxxxxxxxxme",
+      "age": 42,
+      "nicknames" : [ { "name": "xxgo" }, "egon" ],
+      "other": {
+        "name": [ "xxil", { "secret": "superman" } ]
+      }
+      }
+
+`egon` and `superman` are not replaced, because they are not in an
+attribute value of an attribute `name`.
+
+### sub-objects and arrays
+
+If you specify a path and the attribute value is an array then the
+masking decision is applied to each entry of this array as if this
+were the value of the attribute.
+
+If the attribute value is a object, then the attribute is not masked.
+Instead the sub-object is checked further for leaf attributes.
+
+#### Example
+
+Masking `email` will convert
 
-This masking replaces characters with `x` and ` `. Alphanumeric characters,
-`_` and `-` are replaced by `x`, everything else is replaced by ` `.
+
+  { 
+    "email" : "email address" 
+  }
+
+into
+
+  { 
+    "email" : "xxil xxxxxxss" 
+  }
+
+because `email` is a leaf sttribute.
```

Comment:
```suggestion
… because `email` is a leaf attribute.
```

Reply:
@Simran-B Do you mean "put three dots in front"?

Reply:
It's an ellipsis symbol. I like the following style:

```
For example:

    console.log("foo");

… does this and that.
```

The colon indicates that what follows belongs to this sentence. And the ellipsis signifies that the sentence continues after the code example.

Reply:
understood